### PR TITLE
PHP-CS-FIXER and PHPStan

### LINKED ISSUE #7: Fix : workflow status badge
The workflow badge no longer works in the readme (following modifications).
The action is simple.

[Documentation: Adding a workflow status badge](https://docs.github.com/en/actions/managing-workflow-runs/adding-a-workflow-status-badge)

### PR DESCRIPTION
**Pull request by Spomky**
This PR contains opiniated configurations for php-cs-fixer and phpstan.
The current configuration for php-cs-fixer is limited to PSR2 rules. This PR adds a configuration file for php-cs-fixer and adds several rules you may disagree with (just ask if so!).
In addition, this PR also adds a configuration file for PHPStan. Reached [level is 8](https://phpstan.org/user-guide/rule-levels), which is the maximum at the time of writting.

Fixes #7 and introduces a new badge

### CODE REVIEW

**Comment by ismail1432** [comment]
Hey @Spomky 
Thanks for this one !
TBH I am a bit confused for this level, I use this level at a daily basis and from my POV level 8 can be harder for newcomers, I like the idea of this bundle (straightforward and not too complex).
It let the possibility to anyone to contribute to OSS without having a "deep" experience in PHP/Symfony...
I totally agree that level 8 prevents of may bugs and help to prevent BC but it can also add some complexity for beginners/newcomers that want to contribute **BUT** we can also keep this level and keep in mind that some contributors can require our support to fix PHPStan errors

My 2 cents

**Comment by Spomky** [comment]
Hi @ismail1432,

Thank you for being honest.
I agree with you; such a high level will undoubtedly generate frustration for newcomers (error messages may be hard to understand).
I will decrease the level 5 or 6 that are much more simple to apprehend.

**Review by liorchamla — APPROVED** [review]
I dont know what level you want to chose but I approve this PR since in the codebase I did not find :
- dependency in composer.json to install tools
- configuration files out of the box to run fixers

So I approve this as now I will be able easily to fix my next PRs (if there are some)

**Comment by yoanbernabeu** [comment]
I am ok for a level 5 or 6. It is accessible for the beginner (that I am 👌)

File: tests/Functional/AirtableClientTest.php [hunk 1]
```diff
@@ -8,15 +8,21 @@
...
     {
         $kernel = self::bootKernel();
         $container = $kernel->getContainer();
 
-        $this->assertTrue($container->has(AirtableClientInterface::class));
-        $this->assertTrue($container->has("airtable_client"));
-        $this->assertInstanceOf(AirtableClient::class, $container->get("airtable_client"));
+        static::assertTrue($container->has(AirtableClientInterface::class));
+        static::assertTrue($container->has('airtable_client'));
+        static::assertInstanceOf(AirtableClient::class, $container->get('airtable_client'));
```
Comment (TBoileau) [inline tests/Functional/AirtableClientTest.php:26]: What the difference between using the context and static ?

File: src/AirtableClient.php [hunk 2]
```diff
@@ -69,21 +69,21 @@ public function findOneById(string $table, string $id, ?string $dataClass = null
...
+     * {@inheritdoc}
      */
     public function findTheLatest(string $table, $field, ?string $dataClass = null): ?AirtableRecord
     {
-        $url = $this->id . '/'
-            . $table . '?pageSize=1&sort%5B0%5D%5Bfield%5D='
-            . $field . '&sort%5B0%5D%5Bdirection%5D=desc';
+        $url = $this->id.'/'
+            .$table.'?pageSize=1&sort%5B0%5D%5Bfield%5D='
+            .$field.'&sort%5B0%5D%5Bdirection%5D=desc';
```
Comment (TBoileau) [inline src/AirtableClient.php]: Use sprintf instead of concatenation ?

**Review by TBoileau — COMMENTED** [review]
Nice one

File: src/AirtableClient.php (hunk 2 above)
Comment (liorchamla) [reply src/AirtableClient.php]: It was done in my first PR #1 dont understand how we lost that :p

Comment (TBoileau) [reply src/AirtableClient.php]: Weird

File: src/AirtableClient.php [hunk 3]
```diff
@@ -81,9 +81,21 @@ public function findOneById(string $table, string $id, ?string $dataClass = null
...
+                    'field' => $field,
+                    'direction' => 'desc',
+                ],
+            ],
+        ];
+        $url = sprintf(
+            '%s/%s?%s',
+            $this->id,
+            $table,
+            http_build_query($params)
```
Comment (liorchamla) [inline src/AirtableClient.php:97]: Great !

File: src/AirtableClient.php (hunk 2 above)
Comment (Spomky) [reply src/AirtableClient.php]: Actually these lines were not changed in #1.
I updated the PR with another approach using `http_build_query` that should improve readability.

Comment (liorchamla) [reply src/AirtableClient.php]: It was done here : https://github.com/yoanbernabeu/Airtable-Client-Bundle/pull/1/files#diff-4fca284f12d70bb5aeae63b1d3090a19d7d2a047684d5f441cb0d07e41062c24R37

Comment (liorchamla) [reply src/AirtableClient.php]: Anyway the new way u did it is great :)

File: tests/Functional/AirtableClientTest.php (hunk 1 above)
Comment (Spomky) [reply tests/Functional/AirtableClientTest.php:26]: There is no real difference.
The change comes from the rule [`'php_unit_test_case_static_method_calls' => true`](https://github.com/yoanbernabeu/Airtable-Client-Bundle/pull/8/files#diff-8ec5bae185dc9af5b5bfdcd5f1dcbee62f18525ee488996d52b22729f4076ef0R50).
I can revert that change if you prefer the previous call.

Comment (ismail1432) [reply tests/Functional/AirtableClientTest.php:26]: Yes there's no difference.
https://phpunit.de/manual/6.5/en/appendixes.assertions.html#appendixes.assertions.static-vs-non-static-usage-of-assertion-methods

I read that static method in PHPUnit exists to allows developers to use these methods without extending TestCase like an assert library.

**Review by ismail1432 — APPROVED** [review]

**Review by TBoileau — APPROVED** [review]

File: tests/Functional/AirtableClientTest.php (hunk 1 above)
Comment (TBoileau) [reply tests/Functional/AirtableClientTest.php:26]: Ok thanks ;)

**Pull request merged**